clis/teliod/src/main.rs

```diff
@@ -115,14 +161,193 @@ impl CommandListener {
     }
 }
 
+async fn init_api(auth_token: &str) -> Result<DeviceIdentity, TeliodError> {
+    let mut identity_path = dirs::data_local_dir().ok_or(TeliodError::NoDataLocalDir)?;
+    identity_path.push("teliod");
+    if !identity_path.exists() {
+        let _ = fs::create_dir_all(&identity_path);
+    }
+    identity_path.push(&format!("utun10.json"));
+
+    let mut device_identity = DeviceIdentity::new(&identity_path).await;
+
+    let backoff_bounds = ExponentialBackoffBounds {
+        initial: Duration::from_secs(1),
+        maximal: Some(Duration::from_secs(300)),
+    };
+    let mut backoff = ExponentialBackoff::new(backoff_bounds)?;
+
+    if device_identity.machine_identifier.eq("") {
+        device_identity.machine_identifier = match load_identifier_with_exp_backoff(
+            auth_token,
+            device_identity.public_key,
+            backoff.clone(),
```

Comment:
Is `load_identifier_with_exp_backoff` is modifying a shared state of `backoff`? Shouldn't it be taken as `&mut` then? or is this cloning a reference to `backoff`?

Reply:
This cloning is not a reference to the `backoff` but a clone of backoff as we want each backoff to be independent. I mean, if it is a reference and `load_identifier` takes some retires, then the next backoff that `register_machine` gets would follow up right ?

Reply:
Yes exactly, in that case if `load_identifier` only needs an independent instance of `backoff`, that lives only through-out it's scope, then why do we need to pass it a `backoff` at all? Maybe in that case load_identifier should just create it's own instance of an `ExponentialBackoff`?

What do you think? We could have a sane `Default` implementation for ExponentialBackoff, or just a helper function that creates one? or if we want it to be configurable, we could just pass the `max_duration: usize` for example instead of passing `mut backoff: ExponentialBackoff` which would be a bit more straight forward in terms of ownership 😅

Reply:
I think a helper function to create a default backoff sounds nicer!